src/main/java/duke/helper/DateTimeHelper.java

```diff
@@ -0,0 +1,112 @@
+package duke.helper;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+
+public class DateTimeHelper {
+    private String dateTime;
+    private LocalDate fullDate;
+    private String fullTime = "";
+    private String dateStr = "";
+    private String timeStr = "";
+    public DateTimeHelper(String deadline) {
+        this.dateTime = deadline;
+    }
+
+    /**
+     * Format time as ab:cd pm/am
+     * @return time
+     */
+    public String processTimeStr() {
+        try {
+            int time = Integer.parseInt(timeStr);
+            int hours = (time / 100) % 12;
+            int minutes = time % 100;
+
+            String identifier = time / 100 > 12 ? "pm" : "am";
+            if (time / 100 == 12) {
+                hours = 12;
+                identifier = "pm";
+            }
+            else if (time / 100 == 24)
+                identifier = "am";
+
+            String paddingForHour = hours < 10 ? "0" : "";
+            String paddingForMinute = minutes < 10 ? "0" : "";
+            this.fullTime = paddingForHour + hours + ":" + paddingForMinute + minutes + identifier;
+            return "success";
+        } catch (Exception e) {
+            return "error";
+        }
+    }
+
+    /**
+     * Parse date string
+     * @return date
+     */
+    public String processDateStr() {
+        try {
+            this.fullDate = LocalDate.parse(dateStr);
+            return "success";
+        } catch (Exception e) {
+            return "error";
+        }
+    }
+
+    /**
+     * Obtains date and time strings from tokens
+     * @return status message
+     */
+    public String processInput() {
```

Comment:
All the above methods are `process` something but it would be clearer if the method name describes what the method is doing more specifically.